### PR TITLE
Added more fields to fulladleModuleConfig

### PR DESCRIPTION
**Pull request by asadsalman**
This PR adds the ability to override certain fields for modules when using Fulladle. 
You can override the following properties for any module:

1. maxTestShards
2. clientDetails
3. environmentVariables
2. debugApk

### CODE REVIEW

File: fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  /**
+   * When set to false, Fulladle will not automatically add this module to additionalTestApks.
+   *
+   * Default: true
+   */
+  val enabled: Property<Boolean> = objects.property<Boolean>().convention(true)
+
+  /**
+   * The maximum number of shards
```
Comment (runningcode) [inline fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt:20]:
```suggestion
   * The maximum number of shards to be used for this specific test apk.
```

File: fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   */
+  val maxTestShards: Property<Int> = objects.property<Int>().convention(null as Int?)
+
+  /**
+   * A key-value map of additional details to attach to the test matrix.
```
Comment (runningcode) [inline fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt:25]:
```suggestion
   * A key-value map of additional details to attach to the test matrix results file.
```

File: fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  val clientDetails: MapProperty<String, String> = objects.mapProperty()
+
+  /**
+   * The environment variables are mirrored as extra options to the am instrument -e KEY1 VALUE1 … command and
+   * passed to your test runner (typically AndroidJUnitRunner)
+   */
+  val environmentVariables: MapProperty<String, String> = objects.mapProperty()
+
+  /**
+   * the app under test
```
Comment (runningcode) [inline fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt:39]: is this a fully qualified path to the apk under test?

File: fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt [hunk 4]
```diff
@@ -95,3 +122,9 @@ class FulladlePlugin : Plugin<Project> {
     }
   }
 }
+
+fun Map<String, String>.toYaml(indent: Int = 0) = this
```
Comment (runningcode) [inline fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt:126]: Is there a way to re-use some of the code in `YamlWriter.appendMapProperty`?

File: fladle-plugin/src/test/java/com/osacky/flank/gradle/integration/FulladlePluginIntegrationTest.kt [hunk 5]
```diff
@@ -22,11 +22,216 @@ class FulladlePluginIntegrationTest {
...
+       smart-flank-disable-upload: false
+       legacy-junit-result: false
+       full-junit-result: false
+       output-style: single
+      """.trimIndent()
+    )
+  }
+
+    @Test
+    fun fulladleWithSubmoduleOverrides() {
```
Comment (runningcode) [inline fladle-plugin/src/test/java/com/osacky/flank/gradle/integration/FulladlePluginIntegrationTest.kt:127]: very nice! i appreciate the test case here!

File: docs/multi-module-testing.md [hunk 6]
```diff
@@ -53,7 +53,56 @@ Multi module testing can be done by manually specifying [additionalTestApks](/fl
     ./gradlew runFlank
     ```
 
+### Overriding configurations in modules
```
Comment (runningcode) [inline docs/multi-module-testing.md:56]: very nice docs. thank you

**Review by runningcode — COMMENTED** [review]
looks very good. just some nits

File: fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt (hunk 3 above)
Comment (asadsalman) [reply fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladleModuleExtension.kt:39]: what do you mean by fully qualified path? the absolute path?

**Comment by runningcode** [comment]
Could you run `./gradlew :fladle-plugin:formatKotlin` commit and repush? It is failing the ktlint checks.

**Pull request merged**